extensions/aws/src/main/java/io/opentelemetry/extension/aws/AwsXrayPropagator.java

```diff
@@ -228,6 +228,10 @@ private static <C> Context getContextFromHeader(
       return context;
     }
 
+    if (spanId == null || traceId == null) {
+      logger.warning("Both traceId and spanId are required to extract a valid span context. ");
```

Comment:
this seems like it might spam the logs under some circumstances. Perhaps even the normal circumstances if you're running in the environment where this would happen at all. We should definitely not spam the logs in the hot path of people's code.

Reply:
This is to address https://github.com/open-telemetry/opentelemetry-java/issues/4259 issue. 

TLDR: currently users are creating issues expecting that traceId is sufficient (as per XRay behaviour) to createe otel context, which isn't the case. To avoid further confusion, the idea was to add this log warning. Do you have sth better in mind? Perhaps not warning but info level?

Reply:
[ThrottlingLogger](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/common/src/main/java/io/opentelemetry/sdk/internal/ThrottlingLogger.java)?

Reply:
This log message will happen for every request from ELB unfortunately, and with no recourse since ELB actually needs to be updated. ThrottlingLogger may actually be good in addition to a lower log level, but do think we need to not log by default for something with no actionable item.

Reply:
The existing implementation is in the SDK, so not available to propagator implementations. We'd need to put that into some part of the API, which is possible, but I'm not sure where it would live, exactly.

Reply:
Switched to FINEST as suggested by @anuraaga 
Whole change is really only for end user's convenience, so I believe FINEST is a good solution here.